ray-operator/apis/ray/v1/raycluster_types.go

```diff
@@ -121,6 +121,8 @@ type RayClusterStatus struct {
 	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
 	// Status reflects the status of the cluster
+	//
+	// Deprecated: the State field is replaced by the Conditions field.
```

Comment:
Should we include the name of the conditions that replace the existing states here?

Reply:
Or will there be public docs on the new conditions? If so we can just link to that in a follow-up PR

Reply:
Yes, we plan to make a public doc to introduce the new conditions and changes to the .status.state in the following weeks.
<img width="674" alt="image" src="https://github.com/user-attachments/assets/6e564ff3-ef74-4e02-8895-ce48506bb015">